server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -544,20 +546,20 @@ public Clusters(StreamInput in) throws IOException {
             if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_500_053)) {
                 List<Cluster> clusterList = in.readCollectionAsList(Cluster::new);
                 if (clusterList.isEmpty()) {
-                    this.clusterInfo = Collections.emptyMap();
+                    this.clusterInfo = ConcurrentCollections.newConcurrentMap();
                     this.successful = successfulTemp;
                     this.skipped = skippedTemp;
                 } else {
-                    Map<String, AtomicReference<Cluster>> m = new HashMap<>();
-                    clusterList.forEach(c -> m.put(c.getClusterAlias(), new AtomicReference<>(c)));
-                    this.clusterInfo = Collections.unmodifiableMap(m);
+                    Map<String, Cluster> m = new ConcurrentHashMap<>();
```

Comment:
Should this be `ConcurrentCollections.newConcurrentMap();` list you use elsewhere?